omnigan/trainer.py

```diff
@@ -1432,9 +1436,12 @@ def resume(self, inference=False):
         # -----  Single Model Loading  -----
         # ----------------------------------
         else:
-            load_path = Path(self.opts.output_path) / Path(
-                "checkpoints/latest_ckpt.pth"
-            )
+            if self.opts.train.resume_by_pth_path:
+                load_path = Path(self.opts.output_path)
```

Comment:
This won't work, it's not a path to a checkpoint

Reply:
You should add something like 

```python
assert any(
    [
        t in self.opts.tasks
        and self.opts.load_paths[t] != "none"
        and Path(self.opts.load_paths[t]).exists()
        for t in ["m", "p"]
    ]
)
```

and then select either `opts.load_paths.m` or `opts.load_paths.p`

makes sense?

Reply:
@vict0rsch 
What I mean is:

In the training procedure, we have:

```
output_path: /network/tmp1/ccai/checkpoints/tianyu/omnigan/output_dirs/testing
train.resume_by_pth_path: false 
```
in the yaml config file

During inference, if one wants to infer a specific intermediate checkpoint. She/he needs to modify the yaml file to:
```
output_path: /network/tmp1/ccai/checkpoints/tianyu/omnigan/output_dirs/testing (268)/checkpoints/epoch_40_ckpt.pth
train.resume_by_pth_path: true 

```
268 and 40 are just examples for better illustration

Thus, in this way, self.opts.output_path is a path to the checkpoint.

Reply:
> This won't work, it's not a path to a checkpoint

What I mean is:
During training, we have the followings in the config:
```
output_path: /network/tmp1/ccai/checkpoints/tianyu/omnigan/output_dirs/testing
resume_by_pth_path: false
```
But when one wants to resume/infer from a specific pth intermediate checkpoint. She/he needs to modify the config to:
```
output_path: /network/tmp1/ccai/checkpoints/tianyu/omnigan/output_dirs/"testing (268)"/checkpoints/epoch_40_ckpt.pth
resume_by_pth_path: true
```
Thus, it's a path to a checkpoint.

Reply:
I don't agree with what you're saying on the output path. If you look at the code othe things rely on it! It is a path to a directory, not to a checkpoint. That's why I said we should be using `load_paths.m` or `.p`.

Reply:
Also, in software engineering it's always better to catch errors and deal with them when you know they are going to happen instead of letting the system break. That's how you also prevent unexpected side effects